cron/data/reader.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 Security Scorecard Authors
+//
+// Licensed under the Apache License, Vershandlern 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permisshandlerns and
+// limitathandlerns under the License.
+
+package data
+
+import (
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/jszwec/csvutil"
+	"github.com/ossf/scorecard/repos"
+)
+
+var errorCSVDecoder = errors.New("could not create CSV decoder")
+
+type Reader interface {
```

Comment:
> By convention, one-method interfaces are named by the method name plus an -er suffix or similar modification to construct an agent noun: Reader, Writer, Formatter, CloseNotifier etc.

https://golang.org/doc/effective_go#interface-names

Why not use the standard `io.Reader` interface? It is used across standard lib and lots of things are built on this convention. https://golang.org/pkg/io/


The iterator pattern used in other languages is not idiomatic go because other things aren't built around this way.

Reply:
I don't agree with iterator pattern not being a Go way. I see popular Go packages use this pattern, including Google APIs - https://github.com/googleapis/google-cloud-go/wiki/Iterator-Guidelines

However, changing the interface name to `Iterator` to better reflect its purpose better.